### DISCUSSION TITLE
More specific event types on native elements

### DISCUSSION THREAD

**Discussion by henribru** [vuejs/language-tools#297] [Q&A] [answered]
Not sure if this is an upstream issue or not, but figured I could ask here. Currently if you use e.g. `@input` on an input element, `$event` is inferred to be an `Event` and not an `InputEvent`. Is this something that could potentially be improved or will we always have to work around it?

**Comment by johnsoncodehk** [comment, marked as answer]
Event type is upstream problem, see https://github.com/johnsoncodehk/volar/issues/222#issuecomment-850937972.

**Reply by henribru** [reply]
Thanks, missed that issue!
